upp/stages/resampling.py

```diff
@@ -175,6 +176,31 @@ def run_on_region(self, components, region):
                 f" Jets are upsampled at most {np.max(c._ups_max):.0f} times"
             )
 
+    def set_auto_sampling_fraction(self):
+        optimal_frac_list = []
+        for c in self.components:
+            if c.is_target(self.config.target):
+                c.check_num_jets(c.num_jets, sampling_frac=1, cuts=c.cuts)
+                log.debug("target component -- no resampling needed")
+            optimal_frac_list.append(c.get_auto_sampling_frac(c.num_jets, cuts=c.cuts))
```

Comment:
do we need to compute the optimal value for the target component?

Reply:
We do not. I will fix that